components/ntp_background_images/browser/ntp_sponsored_rich_media_source_util.cc

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ntp_background_images/browser/ntp_sponsored_rich_media_source_util.h"
+
+#include <string>
+
+#include "base/check.h"
+#include "base/files/file_path.h"
+#include "brave/components/ntp_background_images/browser/ntp_sponsored_images_data.h"
+
+namespace ntp_background_images {
+
+namespace {
+
+std::optional<std::string> MaybeGetChildCreativeSubdirectories(
```

Comment:
are these methods here just to make them testable? It would be better to make them static methods on the class I think

Reply:
This util we be used by `components/ntp_background_images/browser/ntp_*_source.cc`, hopefully in the not too distant future, rather than their current `IsValidPath` implementations.

Reply:
Renamed to `ntp_sponsored_source_util` so that it can be used by both ntp_sponsored_image and ntp_sponsored_rich_media in a followup.